packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -418,6 +430,7 @@ export class ContainerDefinition extends cdk.Construct {
       command: this.props.command,
       cpu: this.props.cpu,
       disableNetworking: this.props.disableNetworking,
+      dependsOn: this.containerDependencies.map(renderContainerDependency),
```

Comment:
Please don't render the property at all if the list is empty.

Reply:
Yeah, I was unsure about this. Rendering an empty list keeps in consistent with `Links`, `MountPoints` and `VolumesFrom` however it results in a big change. 
I'm happy to make it different to the other three. I'm happy to do a further PR to make the other three not output if their list is empty, would you be open to that?

Reply:
Or, leave it as is, and refactor all four as a seperate PR?

Reply:
We prefer not to render empty arrays/objects. Would be great to fix the others as well...

Reply:
Ok, cool. I can fix the others as well as part of this PR.